### PR TITLE
Add Valid command to spawner and make executors check when opening

### PR DESCRIPTION
**Pull request by dadgar**
Add Valid command to spawner and make executors check when opening

### CODE REVIEW

File: client/driver/spawn/spawn.go [hunk 1]
```diff
@@ -285,3 +285,21 @@ func (s *Spawner) readExitCode() (int, error) {
...
+// throught client restarts. If Valid a nil error is returned.
+func (s *Spawner) Valid() error {
+	// If the spawner is still alive, then the task is running and we can wait
+	// on it.
+	if s.Alive() {
+		return nil
+	}
+
+	// The task isn't alive so check that there is a valid exit code file.
+	if _, err := s.readExitCode(); err == nil {
```
Comment (cbednarski) [inline client/driver/spawn/spawn.go]: Do we want to return the error if it's not nil?

Comment (dadgar) [inline client/driver/spawn/spawn.go]: The reason I didn't was I wanted to have a friendly error code in case it gets exposed to the user. Thoughts?

Comment (cbednarski) [inline client/driver/spawn/spawn.go]: Discussed on Slack but this is OK for now. :+1:

Comment (dadgar) [inline client/driver/spawn/spawn.go]: Talked offline! Keeping as is

**Pull request merged**